pkg/adapter/configError.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adapter
+
+import (
+	"fmt"
+	me "github.com/hashicorp/go-multierror"
+)
+
+// ConfigError represents an error encountered while validating a block of configuration state.
+type ConfigError struct {
+	Field      string
+	Underlying error
+}
+
+// Append adds a ConfigError to a multierror. This function is intended
+// to be used by adapter's ValidateConfig method to report errors
+// in configuration. The field parameter indicates the name of the
+// specific configuration field name that is problematic.
+func Append(base *me.Error, field string, format string, args... interface{}) *me.Error {
```

Comment:
can this be written as:

`func Append(base *me.Error, field, format string, args... interface{})`

(remove the unnecessary 'string' that follows field)?